### PR TITLE
Update properties file path in Java XSettings

### PR DESCRIPTION
**Pull request by shinfan**
More context: https://github.com/GoogleCloudPlatform/google-cloud-java/pull/1863

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java [hunk 1]
```diff
@@ -113,6 +114,13 @@ public String getServicePhraseName(Interface service) {
     return Name.upperCamel(service.getSimpleName()).toPhrase();
   }
 
+  /** Artifact ID of the service */
+  public String getArtifactId(Interface service) {
+    List<String> tokens =
+        Splitter.on('.').splitToList(service.getModel().getServiceConfig().getName());
+    return tokens.size() > 0 ? "google-cloud-" + tokens.get(0) : null;
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java]: You are hardcoding the knowledge of google-cloud. This doesn't work for non-google-cloud services. We need to get this from config somehow.

Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java]: Or it's probably better if it is provided at runtime from the library calling the GAPIC client.

Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java]: Done

File: src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java [hunk 2]
```diff
@@ -198,4 +199,14 @@ public String getReleaseAnnotation(ReleaseLevel releaseLevel) {
   public String getBatchingDescriptorConstName(Method method) {
     return inittedConstantName(Name.upperCamel(method.getSimpleName()).join("batching_desc"));
   }
+
+  @Override
+  public String getPackagePath() {
+    List<String> packagePath = Splitter.on(".").splitToList(getPackageName());
+    if (packagePath.size() > 3) {
+      // Remove the "spi.{version}" suffix
+      return Joiner.on("/").join(packagePath.subList(0, packagePath.size() - 2));
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java]: We should check to make sure our assumption is correct (that the package actually has the spi.version suffix), and if not, either throw an exception or do something different.

Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java]: Switched to use indexOf("spi"). Consider done.

**Comment by shinfan** [comment]
PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**